src/rpc.rs

```diff
@@ -126,24 +146,95 @@ impl RpcSol for RpcSolImpl {
     fn get_transaction_count(&self, meta: Self::Metadata) -> Result<u64> {
         meta.request_processor.get_transaction_count()
     }
-    // fn send_transaction(&self, meta: Self::Metadata, to: String, tokens: i64) -> Result<String> {
-    //     let client_keypair = read_keypair(&meta.keypair_location.unwrap()).unwrap();
-    //     let mut client = mk_client(&meta.leader.unwrap());
-    //     let last_id = client.get_last_id();
-    //     let to_pubkey_vec = bs58::decode(to)
-    //         .into_vec()
-    //         .expect("base58-encoded public key");
-    //
-    //     if to_pubkey_vec.len() != mem::size_of::<Pubkey>() {
-    //         Err(Error::invalid_request())
-    //     } else {
-    //         let to_pubkey = Pubkey::new(&to_pubkey_vec);
-    //         let signature = client
-    //             .transfer(tokens, &client_keypair, to_pubkey, &last_id)
-    //             .unwrap();
-    //         Ok(bs58::encode(signature).into_string())
-    //     }
-    // }
+    fn request_airdrop(&self, meta: Self::Metadata, id: String, tokens: u64) -> Result<bool> {
+        let pubkey_vec = bs58::decode(id)
+            .into_vec()
+            .map_err(|_| Error::invalid_request())?;
+        if pubkey_vec.len() != mem::size_of::<Pubkey>() {
+            return Err(Error::invalid_request());
+        }
+        let pubkey = Pubkey::new(&pubkey_vec);
+        let previous_balance = meta
+            .request_processor
+            .get_balance(pubkey)
+            .map_err(|_| Error::internal_error())?;
+        request_airdrop(&meta.drone_addr, &pubkey, tokens).map_err(|_| Error::internal_error())?;
+        let now = Instant::now();
+        let mut balance;
+        loop {
+            balance = meta
+                .request_processor
+                .get_balance(pubkey)
+                .map_err(|_| Error::internal_error())?;
+            if balance > previous_balance {
+                return Ok(true);
+            } else if now.elapsed().as_secs() > 5 {
+                return Err(Error::internal_error());
+            }
+            sleep(Duration::from_millis(100));
+        }
+    }
+    fn send_transaction(
```

Comment:
I'm a bit confused by the send_transaction parameters.  I was expecting it to just accept an array of octets containing the fully-signed transaction. 

eg, the "data":
```rust
let data = serialize(&tx).expect("serialize Transaction in rpc");
```

Reply:
Ah, my mistake. I see now you said that fairly clearly yesterday. I had transaction signature in my head, rather than complete serialized transaction.
Will fix.

Reply:
Makes my code simpler :)
@mvines , I'm returning the signature to be consistent with our wallet CLI (and also emulate Eth), but if you don't think that's necessary, it would save us the deserialize operation.